libp2p/network/swarm.py

```diff
@@ -320,19 +301,21 @@ def register_notifee(self, notifee: INotifee) -> None:
         self.notifees.append(notifee)
 
     def notify_opened_stream(self, stream: INetStream) -> None:
-        asyncio.gather(
-            *[notifee.opened_stream(self, stream) for notifee in self.notifees]
-        )
+        for notifee in self.notifees:
+            self.manager.run_task(notifee.opened_stream, self, stream)
 
     # TODO: `notify_closed_stream`
 
     def notify_connected(self, conn: INetConn) -> None:
-        asyncio.gather(*[notifee.connected(self, conn) for notifee in self.notifees])
+        for notifee in self.notifees:
+            self.manager.run_task(notifee.connected, self, conn)
```

Comment:
It seems odd to me that this method returns before the notifications have been sent (they will only have been queued)...  Is that intentional?

Reply:
Hmm, you're right. I checked out the implementation in [go](https://github.com/libp2p/go-libp2p-swarm/blob/master/swarm.go#L459). All Notifiees should finish before the function exits.